app/models/medium.rb

```diff
@@ -32,15 +32,31 @@ def create_path
     self.src ||= MediaStorage.stored_path(content_type, type, *path_opts)
   end
 
+  def plural_types
+    @plural_types ||= %w(project workflow tutorial field_guide)
+  end
+
+  def plural_types_regex
+    @plural_types_regex ||= /\A(#{plural_types.join("|")})_attached_image\z/
+  end
+
+  def split_type
```

Comment:
My suggestion would be to do it like this:

```ruby
  # TODO: This method is a good argument for converting this into a STI model
  def location
    details = linked_resource_details

    if pluralize_media_type?(details[:media_type])
      "/#{resource.pluralize}/#{linked_id}/#{media_type.pluralize}/#{id}"
    else
      "/#{resource.pluralize}/#{linked_id}/#{media_type}"
    end
  end

  def linked_resource_details
    /\A(?<resource>field_guide|\w+)_(?<media_type>\w+)/i.match(type)
  end

  def pluralize_media_type?(media_type)
    ["attached_image"].include?(media_type)
  end
```

A couple of specs might help us remember that/why we're doing this too.